src/WebJobs.Extensions.DurableTask/Listener/TaskEntityShim.cs

```diff
@@ -254,19 +159,30 @@ public override async Task<string> Execute(OrchestrationContext innerContext, st
             return string.Empty;
         }
 
-        private async Task ProcessRequestAsync(RequestMessage request)
+        public async Task ExecuteBatch()
         {
-            if (request.IsLockMessage)
+            if (this.GetFunctionInfo().IsOutOfProc)
             {
-                this.ProcessLockRequest(request);
+                // process all operations in the batch using a single function call
+                await this.ExecuteOutOfProcBatch();
```

Comment:
>await this.ExecuteOutOfProcBatch(); [](start = 16, length = 35)

If I understand correctly, in .NET we will always have one function execution per operation, but in non-.NET (e.g. JavaScript) there will be one function execution for the entire batch. Is that correct? If so, it worries me a little from a consistency perspective, especially since that will impact billing, metering/timeouts, tracing, etc. Is there a technical reason we decided to do batching for out-of-proc or would it make sense for us to try and preserve the 1:1 operation to function execution?

Reply:
It was purely a technical reason (tough one can argue that batching is the better design from a performance and billing viewpoint - for example, the latest EventHubs binding also allows batching).

From what I understood we cannot make multiple out-of-proc calls since we cannot pass different parameters to each call as we do in C#... all parameters must be known upfront at the time we call the `Executor.TryExecuteAsync` because that's when they get serialized into the JSON object.